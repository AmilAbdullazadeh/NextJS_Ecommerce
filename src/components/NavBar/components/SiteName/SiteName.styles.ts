import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(({ breakpoints, spacing }) => ({
    siteName: {
        cursor: 'pointer',
        [breakpoints.between(0, 'xs')]: {
            display: 'none',
        },
    },
    siteNameMobile: {
        cursor: 'pointer',
        marginLeft: spacing(-1),
        marginRight: spacing(1),
        [breakpoints.up('sm')]: {
            display: 'none',
        },
    },
}));

export default useStyles;
