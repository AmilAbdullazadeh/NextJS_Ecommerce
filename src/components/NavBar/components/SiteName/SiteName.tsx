import React from 'react';
import { useRouter } from 'next/router';

import { Typography } from '@components';
import { SITE_NAME, landingPage } from '@core';

import useStyles from './SiteName.styles';

const SiteName: React.FC = () => {
    const classes = useStyles();
    const router = useRouter();

    const navigateToLandingPage = () => router.push(landingPage());

    return (
        <>
            <Typography
                component="h1"
                variant="h5"
                color="textSecondary"
                className={classes.siteNameMobile}
                onClick={navigateToLandingPage}
            >
                {SITE_NAME.charAt(0)}
            </Typography>
            <Typography
                component="h1"
                variant="h5"
                color="textSecondary"
                className={classes.siteName}
                onClick={navigateToLandingPage}
            >
                {SITE_NAME}
            </Typography>
        </>
    );
};

export default SiteName;
