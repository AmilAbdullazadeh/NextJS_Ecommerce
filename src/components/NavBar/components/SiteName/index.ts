export { default as SiteName } from './SiteName';
