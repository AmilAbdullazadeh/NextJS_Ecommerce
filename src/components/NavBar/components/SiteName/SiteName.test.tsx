import React from 'react';

import { renderWithProviders } from '@testUtils';

import SiteName from './SiteName';
import { landingPage, SITE_NAME } from '@core';

describe('SiteName', () => {
    afterAll(() => {
        jest.restoreAllMocks();
    });
    it('matches snapshot', () => {
        const { container } = renderWithProviders(<SiteName />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('navigates to landing page', () => {
        // eslint-disable-next-line @typescript-eslint/no-var-requires
        const useRouter = jest.spyOn(require('next/router'), 'useRouter');
        const mockPush = jest.fn();
        useRouter.mockImplementation(() => ({ push: mockPush }));

        const { getByText } = renderWithProviders(<SiteName />);

        const siteName = getByText(SITE_NAME);

        siteName.click();

        expect(mockPush).toHaveBeenLastCalledWith(landingPage());
    });
});
