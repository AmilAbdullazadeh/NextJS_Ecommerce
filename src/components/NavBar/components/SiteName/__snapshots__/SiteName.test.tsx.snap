// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteName matches snapshot 1`] = `
<h1
  class="MuiTypography-root makeStyles-siteNameMobile-2 MuiTypography-h5 MuiTypography-colorTextSecondary"
>
  P
</h1>
`;
